src/doc.c

```diff
@@ -2325,6 +2327,41 @@ void highlightText(Scope *sc, Dsymbol *s, OutBuffer *buf, size_t offset)
                 }
                 break;
 
+            case '`':
+                if (inBacktick)
+                {
+                    inBacktick = 0;
+                    inCode = 0;
+
+                    OutBuffer codebuf;
+
+                    codebuf.write(buf->data + iCodeStart + 1, i - (iCodeStart + 1));
+                    codebuf.writeByte(0);
+
+                    highlightCode2(sc, s, &codebuf, 0);
+
+                    buf->remove(iCodeStart, i - iCodeStart + 1); // also trimming off the current `
+                    i = buf->insert(iCodeStart, codebuf.data, codebuf.offset);
+                    i = buf->insert(i, ")", 1);
+
+                    break;
+                }
+
+                if (inCode)
+                    break;
+
+                inCode = 1;
+                inBacktick = 1;
+
+                static const char pre[] = "$(D_INLINECODE ";
```

Comment:
On Wed, Jan 07, 2015 at 08:05:20AM -0800, Andrei Alexandrescu wrote:
> I don't agree with your disagreement. How is it not syntactic sugar lowered to another macro? This is exactly what it is!

It also calls the highlight code function, transforming the string
before the macro is run.

The text `int < foo`, for example, becomes

$(D_INLINECODE $(D_KEYWORD int) &lt; foo)

The escaping and syntax highlighting wouldn't be done in a regular
macro. (BTW, I could do without syntax highlighting, but the escaping
is an absolute must-have so you can do html code examples in the ticks
and the syntax highlighting is the only function in ddoc that performs
escaping.)